core/src/main/java/org/everit/email/store/ri/EmailStoreImpl.java

```diff
@@ -0,0 +1,688 @@
+    List<Long> binaryContentIds = querydslSupport.execute((connection, configuration) -> {
+      return new SQLQuery(connection, configuration)
+          .from(qInlineImage)
+          .where(qInlineImage.htmlContentId.eq(htmlContentId))
+          .list(qInlineImage.binaryContentId);
+    });
+
+    querydslSupport.execute((connection, configuration) -> {
+      return new SQLDeleteClause(connection, configuration, qInlineImage)
+          .where(qInlineImage.htmlContentId.eq(htmlContentId))
+          .execute();
+    });
+
+    deleteBinaryContents(binaryContentIds);
+  }
+
+  private void deleteRecipients(final long storedEmailId) {
+    QEmailRecipient qEmailRecipient = QEmailRecipient.emailRecipient;
+    List<Long> emailAddressIds = querydslSupport.execute((connection, configuration) -> {
+      return new SQLQuery(connection, configuration)
+          .from(qEmailRecipient)
+          .where(qEmailRecipient.storedEmailId.eq(storedEmailId))
+          .list(qEmailRecipient.emailAddressId);
+    });
+
+    if (emailAddressIds.isEmpty()) {
+      return;
+    }
+
+    querydslSupport.execute((connection, configuration) -> {
+      return new SQLDeleteClause(connection, configuration, qEmailRecipient)
+          .where(qEmailRecipient.storedEmailId.eq(storedEmailId))
+          .execute();
+    });
+
+    deleteEmailAddresses(emailAddressIds);
+  }
+
+  private void deleteTextContent(final long storedEmailId) {
+    QTextContent qTextContent = QTextContent.textContent;
+    Tuple tuple = querydslSupport.execute((connection, configuration) -> {
+      return new SQLQuery(connection, configuration)
+          .from(qTextContent)
+          .where(qTextContent.storedEmailId.eq(storedEmailId))
+          .uniqueResult(qTextContent.textContentId,
+              qTextContent.blobId);
+    });
+
+    if (tuple == null) {
+      return;
+    }
+
+    Long textContentId = tuple.get(qTextContent.textContentId);
+    querydslSupport.execute((connection, configuration) -> {
+      return new SQLDeleteClause(connection, configuration, qTextContent)
+          .where(qTextContent.textContentId.eq(textContentId))
+          .execute();
+    });
+
+    deleteBlob(tuple.get(qTextContent.blobId));
+  }
+
+  private byte[] getBytes(final String content) {
+    if (content == null) {
+      return new byte[0];
+    }
+    return content.getBytes(StandardCharsets.UTF_8);
+  }
+
+  private long insertAddress(final EmailAddress emailAddress) {
+    return querydslSupport.execute((connection, configuration) -> {
+      QAddress qAddress = QAddress.address1;
+      return new SQLInsertClause(connection, configuration, qAddress)
+          .set(qAddress.address, emailAddress.address)
+          .set(qAddress.personal, emailAddress.personal)
+          .executeWithKey(qAddress.emailAddressId);
+    });
+  }
+
+  private void insertAttachments(final Collection<Attachment> attachments,
+      final long storedEmailId) {
+    if (attachments == null) {
+      return;
```

Comment:
Can attachments collection be null? If it is, I would throw an IllegalArgumentException